external/opcua_daq_types/CMakeLists.txt

```diff
@@ -0,0 +1,249 @@
+set_cmake_folder_context(TARGET_FOLDER_NAME)
+project(opcua_daq_types VERSION 1.0.0 DESCRIPTION "DAQ types" LANGUAGES C)
+
+# Define empty list of nodeset dependencies
+list(APPEND OPC_UA_NODESET_DEPENDENCIES)
+
+# Add supported Vendors by openDAQ. Please add the vendor name which is used in this file name pattern:
+# Opc.Ua.Daq.Vendor.xxx.NodeSet2.xml
+list(APPEND OPC_UA_VENDOR_LIST "Hbk")
+# Add repository to be fetched and the used tag. Important index needs to be match with vendor list!
+list(APPEND OPC_UA_VENDOR_REPOSITORY_LIST "https://github.com/hbkworld/opc-ua-specs.git v3.0.12")
+
+# Define empty list for spec directories 
+list(APPEND VENDOR_COMPANION_SPECIFICATIONS_DIRPREFIX)
```

Comment:
For me this is not vendor specific code. This is a vendor specific nodeset to be tested. How else you would guarantee that the openDAQ SDK works with different devices. If someone uses the openDAQ SDK without this option, the user cannot operate with a HBK device - a no go in my perspective.  I would be OK not to include it but then the SDK needs to parse all structs generic.

Reply:
However, there is one more challenge and this why a change the PR into a draft. This implementation requires that the namespace id has a specific value for the hbk vendor nodeset. What happens if another vendor adds only their node set on server side. Then we have a namespace node id missmatch. Client expects for example HBK with namespace 5 and vendor xx with namespace 6.  But on server-side vendor xxx may also use namespace indes 5. This means that we need to have specific namespaces indexes reserved for every vendor spec. I need to check what opc-ua says. Since in theory, every client <-> server connection, need to create through reading out server properties dynamic the used namepsaces with it index.